src/main/java/io/antmedia/datastore/db/MapBasedDataStore.java

```diff
@@ -943,6 +946,79 @@ public boolean deleteToken(String tokenId) {
 		return result;
 	}
 
+	@Override
+	public boolean whiteListToken(String tokenId) {
+		synchronized (this){
+			Token token = getToken(tokenId);
+			if(token != null && token.isBlackListed()){
+				token.setBlackListed(false);
+				return saveToken(token);
+			}
+		}
+
+		return false;
+	}
+
+	@Override
+	public List<String> getBlackListedTokens(){
+		ArrayList<String> tokenBlacklist = new ArrayList<>();
+		synchronized (this){
+			tokenMap.forEach((tokenId, tokenAsJson) -> {
+				Token token = gson.fromJson(tokenAsJson,Token.class);
+				if(token.isBlackListed()){
+					tokenBlacklist.add(tokenId);
+				}
+			});
+			return tokenBlacklist;
+		}
+	}
+
+	@Override
+	public Result deleteAllBlacklistedExpiredTokens(ITokenService tokenService){
+		logger.info("Deleting all expired JWTs from token storage.");
+		AtomicInteger deletedTokenCount = new AtomicInteger();
+
+		synchronized (this) {
+
+			tokenMap.forEach((tokenId, tokenAsJson) -> {
+				Token token = gson.fromJson(tokenAsJson,Token.class);
+				if(token.isBlackListed() && !tokenService.verifyJwt(tokenId,token.getStreamId(),token.getType())){
+					if(deleteToken(tokenId)){
+						deletedTokenCount.getAndIncrement();
+					}else{
+						logger.warn("Couldn't delete JWT:{}", tokenId);
+					}
+				}
+			});
+		}
+
+		if(deletedTokenCount.get() > 0){
+			final String successMsg = deletedTokenCount+" JWT deleted successfully from storage.";
+			logger.info(successMsg);
+			return new Result(true, successMsg);
+		}else{
+			final String failMsg = "No JWT deleted from storage.";
+			logger.warn(failMsg);
+			return new Result(false, failMsg);
+		}
+
+	}
+
+	@Override
+	public boolean whiteListAllTokens(){
+
+		synchronized (this) {
```

Comment:
Using `this` as a synchronization point might not be a good choice as other classes can still hold a lock on this class. I would suggest using a non-public internal locking mechanism for synchronization purposes, which is more fine-grained and only applies to operations on the tokenMap, e.g. a `ReadWriteLock` or similar.